zed/src/main.cpp

```diff
@@ -0,0 +1,238 @@
+
+    // Cam setup
+    InitParameters init_params;
+    init_params.camera_resolution = RESOLUTION::VGA;
+    init_params.coordinate_units = UNIT::METER;
+    init_params.coordinate_system = COORDINATE_SYSTEM::RIGHT_HANDED_Z_UP_X_FWD; // Coordinate system used in ROS
+    init_params.camera_fps = 60;
+    init_params.depth_mode = DEPTH_MODE::NONE;
+		// experimental
+		// init_params.camera_disable_self_calib = false;
+		// init_params.enable_image_enhancement = false;
+
+    // Open the camera
+    ERROR_CODE err = zedm->open(init_params);
+    if (err != ERROR_CODE::SUCCESS) {
+        printf("%s\n", toString(err).c_str());
+        zedm->close();
+    }
+
+    return zedm;
+}
+
+typedef struct {
+	cv::Mat* img0;
+	cv::Mat* img1;
+	std::size_t serial_no;
+} cam_type;
+
+	const record_header __camera_cvtfmt_header {"camera_cvtfmt", {
+		{"iteration_no", typeid(std::size_t)},
+		{"cpu_time_start", typeid(std::chrono::nanoseconds)},
+		{"cpu_time_stop" , typeid(std::chrono::nanoseconds)},
+		{"wall_time_start", typeid(std::chrono::high_resolution_clock::time_point)},
+		{"wall_time_stop" , typeid(std::chrono::high_resolution_clock::time_point)},
+	}};
+
+class zed_camera_thread : public threadloop {
+public:
+  zed_camera_thread(std::string name_, phonebook* pb_, std::shared_ptr<Camera> zedm_)
+  : threadloop{name_, pb_}
+  , sb{pb->lookup_impl<switchboard>()}
+  , _m_cam_type{sb->publish<cam_type>("cam_type")}
+  , zedm{zedm_}
+  , image_size{zedm->getCameraInformation().camera_resolution}
+  , it_log{metric_logger}
+  {
+    // Image setup
+    imageL_zed.alloc(image_size.width, image_size.height, MAT_TYPE::U8_C4, MEM::CPU);
+    imageR_zed.alloc(image_size.width, image_size.height, MAT_TYPE::U8_C4, MEM::CPU);
+
+    imageL_ocv = slMat2cvMat(imageL_zed);
+    imageR_ocv = slMat2cvMat(imageR_zed);
+
+		// experimental
+		runtime_parameters.enable_depth = false;
+  }
+
+private:
+  const std::shared_ptr<switchboard> sb;
+  std::unique_ptr<writer<cam_type>> _m_cam_type;
+  std::shared_ptr<Camera> zedm;
+  Resolution image_size;
+  RuntimeParameters runtime_parameters;
+  std::size_t serial_no {0};
+
+  Mat imageL_zed;
+  Mat imageR_zed;
+
+  cv::Mat imageL_ocv;
+  cv::Mat imageR_ocv;
+  metric_coalescer it_log;
+
+protected:
+	virtual skip_option _p_should_skip() override {
+		if (zedm->grab(runtime_parameters) == ERROR_CODE::SUCCESS) {
+			return skip_option::run;
+		} else {
+			return skip_option::skip_and_spin;
+		}
+	}
+
+  virtual void _p_one_iteration() override {
+      // Retrieve images
+      zedm->retrieveImage(imageL_zed, VIEW::LEFT, MEM::CPU, image_size);
+      zedm->retrieveImage(imageR_zed, VIEW::RIGHT, MEM::CPU, image_size);
+
+	  auto start_cpu_time  = thread_cpu_time();
+	  auto start_wall_time = std::chrono::high_resolution_clock::now();
+
+      // Conversion
+      // cv::cvtColor(imageL_ocv, grayL_ocv, CV_BGR2GRAY);
+      // cv::cvtColor(imageR_ocv, grayR_ocv, CV_BGR2GRAY);
```

Comment:
Shouldn't there be a corresponding change in OV where it does this conversion itself?

Reply:
Reverted.